### PR TITLE
LLM: add GGUF-IQ2 examples

### PR DESCRIPTION
**Pull request by rnwang04**
### 1. Why the change?

https://github.com/intel-analytics/BigDL/pull/10191#issuecomment-1956139882

### 3. Summary of the change 

add GGUF-IQ2 examples

### 4. How to test?
- [x] Local test
- [x] Unit test

### CODE REVIEW

File: python/llm/example/GPU/HF-Transformers-AutoModels/Advanced-Quantizations/2Bit/generate.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                  "or have a cup of coffee now : )")
+
+    # Load model in 2 bit,
+    # which convert the relevant layers in the model into iq2_xxs format.
+    # 2 bit quantization needs weight(imatrix) file to assist in quantization
+    # and improve generation quality, and different model may need different
+    # imtraix file, you can find and download imatrix file from 
+    # https://huggingface.co/datasets/ikawrakow/imatrix-from-wiki-train/tree/main.
+    model = AutoModelForCausalLM.from_pretrained(model_path,
+                                                 load_in_low_bit='iq2_xxs',
```
Comment (jason-dai) [inline python/llm/example/GPU/HF-Transformers-AutoModels/Advanced-Quantizations/2Bit/generate.py:55]: change to `gguf_iq2_xxs`

File: python/llm/example/GPU/HF-Transformers-AutoModels/Advanced-Quantizations/2Bit/README.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
+# 2Bit
```
Comment (jason-dai) [inline python/llm/example/GPU/HF-Transformers-AutoModels/Advanced-Quantizations/2Bit/README.md:1]: Change to `GGUF-IQ2`

Comment (rnwang04) [reply python/llm/example/GPU/HF-Transformers-AutoModels/Advanced-Quantizations/2Bit/README.md]: done

File: python/llm/example/GPU/HF-Transformers-AutoModels/Advanced-Quantizations/2Bit/generate.py (hunk 1 above)
Comment (rnwang04) [reply python/llm/example/GPU/HF-Transformers-AutoModels/Advanced-Quantizations/2Bit/generate.py]: done

File: python/llm/example/GPU/HF-Transformers-AutoModels/Advanced-Quantizations/GGUF-IQ2/README.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
+# GGUF-IQ2
+
+This example shows how to directly run GGUF-IQ2 models using BigDL-LLM on Intel GPU.
```
Comment (jason-dai) [inline python/llm/example/GPU/HF-Transformers-AutoModels/Advanced-Quantizations/GGUF-IQ2/README.md:3]: how to run INT2 models using the IQ2 mechanism (first implemented by llama.cpp) in BigDL-LLM

**Comment by jason-dai** [comment]
See comments inline; otherwise LGTM

File: python/llm/example/GPU/HF-Transformers-AutoModels/Advanced-Quantizations/GGUF-IQ2/README.md (hunk 3 above)
Comment (rnwang04) [reply python/llm/example/GPU/HF-Transformers-AutoModels/Advanced-Quantizations/GGUF-IQ2/README.md]: Have updated to this : )

**Review by MeouSker77 — APPROVED** [review]

**Pull request merged**